src/fields.js

```diff
@@ -910,25 +995,46 @@ x3dom.fields.SFMatrix4f.prototype.toGL = function ()
 };
 
 /**
- * Returns a column major version of this matrix, packed into a single array.
- * @returns {Number[]} resulting array of 16 values
+ * Creates and returns a new <code>SFMatrix4f</code> with the elements,
+ * construed as being in column-major format, copied from the supplied
+ * array.
+ *
+ * @param   {Number[]} array - an array of at least 16 numbers,
+ *                             the first 16 of which will be construed
+ *                             as the new matrix data in column-major
+ *                             format
+ * @returns {SFMatrix4f} a new matrix based upon the <code>array</code>
+ *                       data transferred from its imputed column-major
+ *                       format into the row-major format
  */
 x3dom.fields.SFMatrix4f.fromGL = function ( array )
 {
-    return [
-        this._00, this._10, this._20, this._30,
-        this._01, this._11, this._21, this._31,
-        this._02, this._12, this._22, this._32,
-        this._03, this._13, this._23, this._33
-    ];
+    var newMatrix = new x3dom.fields.SFMatrix4f();
```

Comment:
This would change the method but makes more sense. Need to check the codebase how it is called, eg. with an array argument.

Reply:
apparently never used. So ok.